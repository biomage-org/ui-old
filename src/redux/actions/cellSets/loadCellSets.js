import fetchAPI from 'utils/http/fetchAPI';
import handleError from 'utils/http/handleError';
import {
  CELL_SETS_LOADED, CELL_SETS_LOADING, CELL_SETS_ERROR,
} from 'redux/actionTypes/cellSets';
import endUserMessages from 'utils/endUserMessages';

import config from 'config';
import { api } from 'utils/constants';

const loadCellSets = (experimentId, forceReload = false) => async (dispatch, getState) => {
  const {
    loading, error, updatingClustering,
  } = getState().cellSets;

  if (!forceReload && ((!loading && !error) || updatingClustering)) {
    return null;
  }

  if (error) {
    dispatch({
      type: CELL_SETS_LOADING,
    });
  }

  let url;

  if (config.currentApiVersion === api.V1) {
    url = `/v1/experiments/${experimentId}/cellSets`;
  } else if (config.currentApiVersion === api.V2) {
    url = `/v2/experiments/${experimentId}/cellSets`;
  }

  try {
    const data = await fetchAPI(url);

    dispatch({
      type: CELL_SETS_LOADED,
      payload: {
        experimentId,
        data: data.cellSets,
      },
    });
  } catch (e) {
    const errorMessage = handleError(e, endUserMessages.ERROR_FETCHING_CELL_SETS);

    dispatch({
      type: CELL_SETS_ERROR,
      payload: { error: errorMessage },
    });
  }
};

export default loadCellSets;
