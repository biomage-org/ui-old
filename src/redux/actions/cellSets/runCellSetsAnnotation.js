import {
  CELL_SETS_ERROR, CELL_SETS_CLUSTERING_UPDATING,
} from 'redux/actionTypes/cellSets';

import getTimeoutForWorkerTask from 'utils/getTimeoutForWorkerTask';
import fetchWork from 'utils/work/fetchWork';

const runCellSetsAnnotation = (experimentId, species, tissue) => async (dispatch, getState) => {
  const { error, updatingClustering, loading } = getState().cellSets;

  if ((loading && updatingClustering) || error) return;

  const body = {
    name: 'ScTypeAnnotate',
    species,
    tissue,
  };

  dispatch({
    type: CELL_SETS_CLUSTERING_UPDATING,
  });

  const timeout = getTimeoutForWorkerTask(getState(), 'ClusterCells');

  try {
    await fetchWork(
      experimentId,
      body,
      getState,
      dispatch,
      {
        timeout,
        broadcast: true,
      },
    );
  } catch (e) {
    dispatch({
      type: CELL_SETS_ERROR,
      payload: {
        experimentId,
        error: e,
      },
    });
  }
};

export default runCellSetsAnnotation;
