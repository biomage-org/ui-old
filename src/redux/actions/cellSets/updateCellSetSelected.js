import { CELL_SETS_SET_SELECTED } from '../../actionTypes/cellSets';

const updateCellSetSelected = (keys) => async (dispatch, getState) => {
  const {
    loading, error,
  } = getState().cellSets;

  if (loading || error) {
    return null;
  }

  await dispatch({
    type: CELL_SETS_SET_SELECTED,
    payload: {
      keys,
    },
  });
};

export default updateCellSetSelected;
