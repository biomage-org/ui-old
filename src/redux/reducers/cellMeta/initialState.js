const mitochondrialContent = {
  loading: true,
  error: false,
  data: [],
};

const doubletScores = {
  loading: true,
  error: false,
  data: [],
};

const numOfGenes = {
  loading: true,
  error: false,
  data: [],
};

const numOfUmis = {
  loading: true,
  error: false,
  data: [],
};

const initialState = {
  mitochondrialContent,
  doubletScores,
  numOfGenes,
  numOfUmis,
};

export default initialState;
