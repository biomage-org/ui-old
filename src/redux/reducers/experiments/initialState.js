const experimentTemplate = {
  name: null,
  description: 'Analysis description',
  id: null,
  createdAt: null,
  updatedAt: null,
  notifyByEmail: true,
  meta: { organism: null, type: '10x' },
  sampleIds: [],
  metadataKeys: [],
};

const initialState = {
  meta: {
    loading: false,
    error: false,
    saving: false,
    activeExperimentId: null,
  },
  ids: [],
};

const METADATA_DEFAULT_VALUE = 'N.A.';

export default initialState;
export { experimentTemplate, METADATA_DEFAULT_VALUE };
