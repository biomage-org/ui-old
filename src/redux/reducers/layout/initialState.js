const initialState = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Cell sets and Metadata', direction: 'row', splitPercentage: 60,
      },
      second: 'Heatmap',
      direction: 'column',
      splitPercentage: 50,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

export default initialState;
