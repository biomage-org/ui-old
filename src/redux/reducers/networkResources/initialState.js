const initialState = {
  environment: undefined,
  domainName: undefined,
};

export default initialState;
