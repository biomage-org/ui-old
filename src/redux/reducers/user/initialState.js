const initialState = {
  current: null,
};

export default initialState;
