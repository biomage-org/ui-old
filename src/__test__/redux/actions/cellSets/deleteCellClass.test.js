import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

import waitForActions from 'redux-mock-store-await-actions';

import { CELL_CLASS_DELETE } from 'redux/actionTypes/cellSets';
import initialState from 'redux/reducers/cellSets/initialState';
import deleteCellClass from 'redux/actions/cellSets/deleteCellClass';

enableFetchMocks();

const mockStore = configureStore([thunk]);

describe('deleteCellClass action', () => {
  const experimentId = '1234';
  const key = 'my-key';

  beforeEach(() => {
    const response = new Response(JSON.stringify({}));

    fetchMock.resetMocks();
    fetchMock.doMock();
    fetchMock.mockResolvedValueOnce(response);
  });

  it('Does not dispatch on loading state', async () => {
    const store = mockStore({ cellSets: { loading: true, error: false } });
    store.dispatch(deleteCellClass(experimentId, key));
    expect(store.getActions().length).toEqual(0);
  });

  it('Does not dispatch on error state', async () => {
    const store = mockStore({ cellSets: { loading: false, error: true } });
    store.dispatch(deleteCellClass(experimentId, key));
    expect(store.getActions().length).toEqual(0);
  });

  it('Dispatches an action to delete cell class to the reducer', async () => {
    const store = mockStore({ cellSets: { ...initialState, loading: false } });
    store.dispatch(deleteCellClass(experimentId, key));

    await waitForActions(store, [CELL_CLASS_DELETE]);

    const firstAction = store.getActions()[0];

    expect(firstAction).toMatchSnapshot();
  });

  it('Sends fetch to the API when a cell set is deleted', async () => {
    const store = mockStore({ cellSets: { ...initialState, loading: false } });
    await store.dispatch(deleteCellClass(experimentId, key));

    expect(fetch).toHaveBeenCalledTimes(1);

    const [url, body] = fetch.mock.calls[0];

    expect(url).toEqual('http://localhost:3000/v2/experiments/1234/cellSets');
    expect(body).toMatchSnapshot();
  });

  it('Uses V2 URL when using API version V2', async () => {
    const store = mockStore({ cellSets: { ...initialState, loading: false } });
    await store.dispatch(deleteCellClass(experimentId, key));

    expect(fetchMock).toHaveBeenCalledTimes(1);

    const [url, body] = fetch.mock.calls[0];

    expect(url).toEqual('http://localhost:3000/v2/experiments/1234/cellSets');
    expect(body).toMatchSnapshot();
  });
});
