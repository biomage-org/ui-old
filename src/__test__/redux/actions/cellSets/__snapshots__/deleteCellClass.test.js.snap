// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteCellClass action Dispatches an action to delete cell class to the reducer 1`] = `
{
  "payload": {
    "key": "my-key",
  },
  "type": "cellSets/deleteClass",
}
`;

exports[`deleteCellClass action Sends fetch to the API when a cell set is deleted 1`] = `
{
  "body": "[{"$match":{"query":"$[?(@.key == \\"my-key\\")]","value":{"$remove":true}}}]",
  "headers": {
    "Content-Type": "application/boschni-json-merger+json",
  },
  "method": "PATCH",
}
`;

exports[`deleteCellClass action Uses V2 URL when using API version V2 1`] = `
{
  "body": "[{"$match":{"query":"$[?(@.key == \\"my-key\\")]","value":{"$remove":true}}}]",
  "headers": {
    "Content-Type": "application/boschni-json-merger+json",
  },
  "method": "PATCH",
}
`;
